ansible/roles/kubernetes/defaults/main.yml

```diff
@@ -1,10 +1,9 @@
 ---
-kubernetes:
-  version: "1.9.2-00"
-  cni_version: "0.6.0-00"
-  enable_cached_images: False
-  cached_images: []
-  apt_key_url: "https://packages.cloud.google.com/apt/doc/apt-key.gpg"
-  apt_repo_string: "deb http://apt.kubernetes.io/ kubernetes-{{ ansible_distribution_release|lower }} main"
-  yum_baseurl: "https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64"
-  yum_gpgkey: "https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg"
+kubernetes_version: "{{ kubernetes.version if kubernetes_version_defined else '1.9.3-00' }}"
```

Comment:
where would I set `kubernetes.version`? Is this written this way for backwards compatibility so it doesn't break if you provide a JSON object instead of directly setting these values (like you do with `kubernetes_version` later on)?

Reply:
Correct, this and the associated kubernetes_*_defined vars are simply to provide backward compatibility to the dict-based vars that were previously used.